File: app/src/main/java/com/javinezpaul/gradeportalschool/TeacherMainScreen.java
package com.javinezpaul.gradeportalschool;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;

import java.util.ArrayList;

public class TeacherMainScreen extends AppCompatActivity {
    private Button logout,newcode;
    private TextView teacheruser;

    private RecyclerView teachersCodeRecView;
    private TeachersCodeRecViewAdapter adapter;

    private String schoolcode, userid;

    ArrayList<TeachersCode> teachersCodes = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_teacher_main_screen);

        SharedPreferences sp = getSharedPreferences("credentials",MODE_PRIVATE);
        schoolcode= (sp.getString("schoolcode",""));
        userid= (sp.getString("user",""));
        if(sp.contains("schoolcode")){
            schoolcode= (sp.getString("schoolcode",""));
        }
        if(sp.contains("user")){
            userid= (sp.getString("user",""));
        }

        Toast.makeText(getApplicationContext(),"schoolcode: " + schoolcode,Toast.LENGTH_LONG).show();

        teacheruser = (TextView) findViewById(R.id.teacheruser);
        logout = (Button) findViewById(R.id.logout);
        newcode = (Button) findViewById(R.id.newcode);

        //TODO: insert lines 48 - 67 in volley request function.
        // RecView Adapter and initialization
        teachersCodeRecView = findViewById(R.id.teachersCodeRecView);
        adapter = new TeachersCodeRecViewAdapter(TeacherMainScreen.this);
        teachersCodeRecView.setAdapter(adapter);
        teachersCodeRecView.setLayoutManager(new LinearLayoutManager(TeacherMainScreen.this));

        teachersCodes.clear();

        showTeacherscode();

        //adding data in recyclerview
//        teachersCodes.clear();
//        teachersCodes.add(new TeachersCode("1", "TE1234", "1", "1", "1", "1",
//                        "1", "2021-02-20", "IIT", "Industrial and Information Technology", "ACT",
//                "Associate in Computer Technology", "IT1A", "GENSCI", "General Science 101",
//                "2020", "2021", "1st", "1st"));
//        teachersCodes.add(new TeachersCode("1", "TE4321", "1", "1", "1", "1",
//                "1", "2021-02-20", "IIT", "Industrial and Information Technology", "ACT",
//                "Associate in Computer Technology", "IT1A", "GENSCI", "Physical Science",
//                "2020", "2021", "2nd", "2nd"));
//
//        adapter.setTeachersCodes(teachersCodes);









//        logout.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                SharedPreferences sp = getSharedPreferences("credentials",MODE_PRIVATE);
//                if(sp.contains("user")){
//                    SharedPreferences.Editor editor = sp.edit();
//                    editor.remove("user");
//                    editor.remove("cardid");
//                    editor.remove("name");
//                    editor.remove("image");
//                    editor.remove("access");
//                    editor.putString("msg","Logged Out Successfully");
//                    editor.commit();
//                    Intent hasloggedout = new Intent(TeacherMainScreen.this , Login.class);
//                    startActivity(hasloggedout);
//                    finish();
//                }
//
//            }
//        });
        // session teacher user

        newcode.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(TeacherMainScreen.this,newcode.class);
                startActivity(i);
            }
        });




        // has shown data teacher main screen

    }

    @Override
    protected void onRestart() {
        super.onRestart();
        showTeacherscode();
    }

    //3 dots menu
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.teacher_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()){
            case R.id.Profile:
                getProfile();
                return true;
//            case R.id.Students:
//                Toast.makeText(this,"VIEW CODE SELECTED",Toast.LENGTH_SHORT).show();
//                return true;
            case R.id.about:
                Toast.makeText(this,"About Page",Toast.LENGTH_SHORT).show();
                Intent about = new Intent(TeacherMainScreen.this,About.class);
                startActivity(about);
                return true;
            case R.id.Logout:
               // Toast.makeText(this,"VIEW CODE SELECTED",Toast.LENGTH_SHORT).show();
                SharedPreferences sp = getSharedPreferences("credentials",MODE_PRIVATE);
                if(sp.contains("user")){
                    SharedPreferences.Editor editor = sp.edit();
                    editor.remove("user");
                    editor.remove("cardid");
                    editor.remove("name");
                    editor.remove("image");
                    editor.remove("access");
                    editor.remove("userid");
                    editor.clear();
                    editor.putString("msg","Logged Out Successfully");
                    editor.commit();
                    Intent hasloggedout = new Intent(TeacherMainScreen.this , Login.class);
                    startActivity(hasloggedout);
                    finish();
                }
                return true;
        }
        return super.onOptionsItemSelected(item);
    }


    @Override
    public void onBackPressed(){
        //Toast.makeText(getApplicationContext(),"Press logout button", Toast.LENGTH_LONG).show();
        return;

    }

    private void getProfile() {
        RequestQueue queueu = Volley.newRequestQueue(this);
        String url = "http://jeepcard.net/gportal/profile.php?cardid="+userid;
        StringRequest stringRequest = new StringRequest(Request.Method.GET, url, new Response.Listener<String>() {
            @Override
            public void onResponse(String response) {
//                collegeCounterTextView.setText("Response Get: "+response);
//                Toast.makeText(getApplicationContext(), response, Toast.LENGTH_LONG).show();
                try {
                    JSONArray JA= new JSONArray(response);

                    AlertDialog.Builder builder = new AlertDialog.Builder(TeacherMainScreen.this);
                    builder.setTitle("Profile");
                    builder.setMessage("\nCollege: " + JA.getJSONObject(0).get("collegecode").toString() + "\n" +
                            "Employee ID: " + JA.getJSONObject(0).get("cardid").toString() + "\n\n" +
                            JA.getJSONObject(0).get("lname").toString() + ", " +
                            JA.getJSONObject(0).get("fname").toString() + " " +
                            JA.getJSONObject(0).get("mname").toString() + "\n" +
                            JA.getJSONObject(0).get("gender").toString()
                    );
                    builder.setNegativeButton("Ok", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            dialog.cancel();
                        }
                    });

                    builder.show();
                } catch (JSONException e) {
                    e.printStackTrace();
                }

            } //onResponse
        }, new Response.ErrorListener(){
            @Override
            public void onErrorResponse(VolleyError error) {
                Toast.makeText(getApplicationContext(), "Network unstable, please retry", Toast.LENGTH_LONG).show();
            }
        });//Stringrequest
        queueu.add(stringRequest);
    }

    public void showTeacherscode(){
        // session teacher user
        SharedPreferences sp = getSharedPreferences("credentials",MODE_PRIVATE);
        // volley data
        String getcardidTeacher = (String) sp.getString("user","");
        RequestQueue requestQueue = Volley.newRequestQueue(TeacherMainScreen.this);
        String url = "http://jeepcard.net/gportal/teachermainscreen.php?cardid="+getcardidTeacher;
        StringRequest stringRequest = new StringRequest(Request.Method.GET, url, new Response.Listener<String>() {
            @Override
            public void onResponse(String response) {
                //Toast.makeText(getApplicationContext(),response,Toast.LENGTH_LONG).show();
                teachersCodes.clear();
                try {
                    JSONArray jsonArray = new JSONArray(response);
                    for(int i=0;i<jsonArray.length();i++){
                        String teacherscodeid=jsonArray.getJSONObject(i).get("teacherscodeid").toString();
                        String teacherscode=jsonArray.getJSONObject(i).get("teacherscode").toString();
                        String collegeid=jsonArray.getJSONObject(i).get("collegeid").toString();
                        String programid=jsonArray.getJSONObject(i).get("programid").toString();
                        String sectionid=jsonArray.getJSONObject(i).get("sectionid").toString();
                        String subjectid=jsonArray.getJSONObject(i).get("subjectid").toString();
                        String ayid=jsonArray.getJSONObject(i).get("ayid").toString();
                        String userid=jsonArray.getJSONObject(i).get("userid").toString();
                        String date=jsonArray.getJSONObject(i).get("date").toString();
                        String status=jsonArray.getJSONObject(i).get("status").toString();
                        String collegecode=jsonArray.getJSONObject(i).get("collegecode").toString();
                        String programcode=jsonArray.getJSONObject(i).get("programcode").toString();
                        String sectioncode=jsonArray.getJSONObject(i).get("sectioncode").toString();
                        String subjectcode=jsonArray.getJSONObject(i).get("subjectcode").toString();
                        String collegename=jsonArray.getJSONObject(i).get("collegename").toString();
                        String programname=jsonArray.getJSONObject(i).get("programname").toString();
                        String subjecttitle=jsonArray.getJSONObject(i).get("subjecttitle").toString();
                        String ayearlevel=jsonArray.getJSONObject(i).get("ayearlevel").toString();
                        String aysem=jsonArray.getJSONObject(i).get("aysem").toString();
                        String ayear1=jsonArray.getJSONObject(i).get("ayear1").toString();
                        String ayear2=jsonArray.getJSONObject(i).get("ayear2").toString();
                        String fullname=jsonArray.getJSONObject(i).get("fullname").toString();
                        String access=jsonArray.getJSONObject(i).get("access").toString();
                        String photo=jsonArray.getJSONObject(i).get("photo").toString();
                        String cardid=jsonArray.getJSONObject(i).get("cardid").toString();

                        teachersCodes.add(new TeachersCode(teacherscodeid, teacherscode, collegeid, programid, sectionid, subjectid,
                                ayid, date, collegecode, collegename, programcode,
                                programname, sectioncode, subjectcode, subjecttitle,
                                ayear1, ayear2, ayearlevel, aysem));
                        adapter.setTeachersCodes(teachersCodes);

                        if(sp.contains("user")){
                            SharedPreferences.Editor editor = sp.edit();
                            editor.putString("name",fullname);
                            editor.putString("cardid",cardid);
                            editor.putString("image",photo);
                            editor.putString("access",access);
                            editor.putString("userid",userid);
                            editor.commit();
                        }
                    }
                } catch (JSONException e) {
                    e.printStackTrace();
                }

            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Toast.makeText(getApplicationContext(),"Something went wrong!\n"+error.toString(),Toast.LENGTH_LONG).show();
            }
        });
        requestQueue.add(stringRequest);



        //volley data
    }
}

File: phpMysql/teachermainscreen.php
<?php

require './gportal_android_database/database.php';

$pdo = Database::connect();

date_default_timezone_set('Asia/Manila');



//echo "success";

//echo $_GET["cardid"];

$cardid = @$_GET["cardid"];



        $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

        $sqlteacher = "SELECT cardid,userid FROM usertbl WHERE cardid=?";

        $qteacher = $pdo->prepare($sqlteacher);

        $qteacher->execute(array($cardid));

        $cardidFound = $qteacher->fetch(PDO::FETCH_ASSOC);

        //echo $cardid["cardid"];        

        if($cardidFound["cardid"]==$cardid){

             //echo $cardidFound["cardid"];

            $sqlFrom = 'SELECT *, 

            (SELECT collegecode FROM collegetbl WHERE teacherscodetbl.collegeid=collegetbl.collegeid) AS collegecode, 

            (SELECT programcode FROM programtbl WHERE teacherscodetbl.programid=programtbl.programid) AS programcode, 

            (SELECT sectioncode FROM sectiontbl WHERE teacherscodetbl.sectionid=sectiontbl.sectionid) AS sectioncode, 

            (SELECT subjectcode FROM subjecttbl WHERE teacherscodetbl.subjectid=subjecttbl.subjectid) AS subjectcode,
           (SELECT CONCAT(lname,", ",fname," ",mname,".") as fullnane FROM usertbl WHERE teacherscodetbl.userid=usertbl.userid) 
AS fullname,
            (SELECT access FROM usertbl WHERE teacherscodetbl.userid=usertbl.userid) AS access,
            (SELECT photo FROM usertbl WHERE teacherscodetbl.userid=usertbl.userid) AS photo,
            (SELECT cardid FROM usertbl WHERE teacherscodetbl.userid=usertbl.userid) AS cardid,

            (SELECT 

                IF(ayearlevel=1,"1st",

                IF(ayearlevel=2,"2nd",

                IF(ayearlevel=3,"3rd",

                IF(ayearlevel=4,"4th",

                "")))) 

             AS ayearlevel FROM aytbl WHERE teacherscodetbl.ayid=aytbl.ayid)

                AS ayearlevel,

            (SELECT IF(aysem=1,"1st",

                       IF(aysem=2,"2nd",

                          ""))

            AS aysem FROM aytbl WHERE teacherscodetbl.ayid=aytbl.ayid) 

                AS aysem,

            (SELECT ayear1 FROM aytbl WHERE teacherscodetbl.ayid=aytbl.ayid) AS ayear1,

            (SELECT ayear2 FROM aytbl WHERE teacherscodetbl.ayid=aytbl.ayid) AS ayear2,

            (SELECT programname FROM programtbl WHERE teacherscodetbl.programid=programtbl.programid) AS programname,

            (SELECT collegename FROM collegetbl WHERE teacherscodetbl.collegeid=collegetbl.collegeid) AS collegename,

            (SELECT subjecttitle FROM subjecttbl WHERE teacherscodetbl.subjectid=subjecttbl.subjectid) AS subjecttitle

            FROM teacherscodetbl WHERE teacherscodetbl.userid=? AND teacherscodetbl.status="active" 
			ORDER BY subjecttitle';

            $sqlFrom = $pdo->prepare($sqlFrom);

            $sqlFrom->execute(array($cardidFound["userid"]));

            $rowCount = $sqlFrom->rowCount();

            if($rowCount>0){

                $data = array();

                while($rowFrom = $sqlFrom->FETCH(PDO::FETCH_ASSOC)){

                    array_push($data,$rowFrom);

                }

                echo json_encode($data);

            }



        }

         else{

             echo "FAILED";

        }



    // if(@$_GET["valid"]=="true"){

    //     $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

    //     $sqlGetFrom = 'SELECT *, (SELECT subjectcode FROM subjecttbl WHERE teacherscodetbl.subjectid=subjecttbl.subjectid) as subjectcode, (SELECT sectioncode FROM sectiontbl WHERE teacherscodetbl.sectionid=sectiontbl.sectionid) as sectioncode, (SELECT collegecode from collegetbl where teacherscodetbl.collegeid=collegetbl.collegeid) as collegecode, (select programcode from programtbl where teacherscodetbl.programid=programtbl.programid) as programcode, (SELECT concat(ayear1, " - ", ayear2) as ay from aytbl where teacherscodetbl.ayid=aytbl.ayid) as ay FROM teacherscodetbl where teacherscode="te234ths" AND status="active"';

    //     $sqlFrom = $pdo->prepare($sqlGetFrom);



    //         $sqlFrom->execute();

    //         $rowCount = $sqlFrom->rowCount();

    //         if($rowCount>0){

    //             $data = array();

    //             while($rowFrom = $sqlFrom->FETCH(PDO::FETCH_ASSOC)){

    //                 array_push($data,$rowFrom);

    //             }

    //             echo json_encode($data);

    //         }

    // } 

    // else{

    //     echo "No data";

    // } 



?>